modules/ai/assets/js/editor/components/prompt-history/parts/modal-item.js

```diff
@@ -0,0 +1,150 @@
+import {
+	Box,
+	ListItem,
+	ListItemButton,
+	ListItemIcon,
+	ListItemText,
+	Stack,
+	styled,
+} from '@elementor/ui';
+import EditIcon from '../../../icons/edit-icon';
+import { TrashIcon } from '@elementor/icons';
+import RestoreIcon from '../../../icons/restore-icon';
+import ActionButton from './modal-item-action-button';
+import { ACTION_TYPES, usePromptHistoryAction } from '../context/prompt-history-action-context';
+import PromptHistoryActionIcon from './action-icon';
+import RefreshIcon from '../../../icons/refresh-icon';
+import PromptHistoryItemTitle from './modal-item-title';
+import PromptHistoryItemSecondaryContent from './modal-item-secondary-content';
+
+const StyledListItem = styled( ListItem )`
+  & .e-prompt-history-item-actions {
+    position: absolute;
+    top: -9999px;
+  }
+
+  &:hover,
+  &:focus-visible,
+  &:focus-within {
+    .MuiTypography-subtitle1 {
+      font-weight: 700;
+      max-width: ${ ( { actionsCount } ) => `calc(100% - ${ actionsCount } * 26px)` };
+    }
+
+    .e-prompt-history-item-actions {
+      position: relative;
+      top: -3px;
+    }
+  }
+`;
+
+const StyledListItemButton = styled( ListItemButton )`
+  padding: ${ ( { theme } ) => theme.spacing( 0.5, 2 ) };
+  align-items: flex-start;
+  cursor: inherit;
+`;
+
+const StyledListItemIcon = styled( ListItemIcon )`
+  min-width: auto;
+
+  & > .MuiSvgIcon-root {
+    margin-right: ${ ( { theme } ) => theme.spacing( 1.5 ) };
+    font-size: 1rem;
+    position: relative;
+    top: ${ ( { theme } ) => theme.spacing( 0.5 ) };
+  }
+`;
+
+const StyledButtonsWrapper = styled( Box )`
+  & .MuiSvgIcon-root {
+    font-size: 1rem;
+  }
+`;
+
+const PromptHistoryItem = ( {
+	id,
+	date,
+	onHistoryItemDelete,
+	...props
+} ) => {
+	const { getAllowedActions, onPromptReuse, onResultEdit, onImagesRestore } = usePromptHistoryAction();
+	const { action, prompt, text, images, thumbnails, imageType, ratio } = props;
+	const allowedActions = getAllowedActions();
+
+	return (
+		<StyledListItem
+			tabIndex="0"
+			data-testid="e-ph-i"
+			disableGutters={ true }
+			disablePadding={ true }
+			actionsCount={ Object.keys( allowedActions ).length }>
+			<StyledListItemButton component="div" role={ undefined }>
+				<StyledListItemIcon>
+					<PromptHistoryActionIcon action={ action } />
+				</StyledListItemIcon>
+
+				<ListItemText
```

Comment:
General overview:

Instead of holding all the action types together (text, code, images) and all of their logic, and then everywhere to make multiple conditions per each prompt type, why not separate concerns so that each module will handle its own data?

( It's affecting the entire code: by holding all the data handling together: multiple conditions are needed to distinguish between the modules type).

Reply:
Hmm.

Since the history modal behaviour is not dependent on the dialogue we opened it from, why should we actually create that dependency? Now it's fully autonomous and it's complexity won't increase much if we add a new history type (there will be blocks and maybe something else looking forward) or a new action to the list.

So the history component knows only about the history and produces events, other components are listening to these events and do what they should. I believe this approach is scalable and pretty straightforward for the newcomers. 

Correct me if I wrong 🙂

Reply:
As I see it, for example in the text module - thinking about what we have there in terms of the data, we need to load a list of items, and then each action of this item relates to the module itself (edit or reuse the prompt for example).

Therefore we probably need a shared UI component/s that renders the list, and the actions themselves are already at the module level.
So the thing that is really shared here mostly is the UI.. the functionality happens in the module itself anyway.
This is why I've suggested rendering the dialog inside of the module itself.